core/src/main/java/com/google/instrumentation/trace/SpanData.java

```diff
@@ -0,0 +1,278 @@
+   *
+   * @return the parent {@code SpanId} or {@code null} if the {@code Span} is a root {@code Span}.
+   */
+  @Nullable
+  public abstract SpanId getParentSpanId();
+
+  /**
+   * Returns the display name of this {@code Span}.
+   *
+   * @return the display name of this {@code Span}.
+   */
+  public abstract String getDisplayName();
+
+  /**
+   * Returns the start {@code Timestamp} of this {@code Span}.
+   *
+   * @return the start {@code Timestamp} of this {@code Span}.
+   */
+  public abstract Timestamp getStartTimestamp();
+
+  /**
+   * Returns the attributes recorded for this {@code Span}.
+   *
+   * @return the attributes recorded for this {@code Span}.
+   */
+  public abstract Attributes getAttributes();
+
+  /**
+   * Returns the annotations recorded for this {@code Span}.
+   *
+   * @return the annotations recorded for this {@code Span}.
+   */
+  public abstract TimedEvents<Annotation> getAnnotations();
+
+  /**
+   * Returns network events recorded for this {@code Span}.
+   *
+   * @return network events recorded for this {@code Span}.
+   */
+  public abstract TimedEvents<NetworkEvent> getNetworkEvents();
+
+  /**
+   * Returns links recorded for this {@code Span}.
+   *
+   * @return links recorded for this {@code Span}.
+   */
+  public abstract Links getLinks();
+
+  /**
+   * Returns the {@code Status} or {@code null} if {@code Span} is still active.
+   *
+   * @return the {@code Status} or {@code null} if {@code Span} is still active.
+   */
+  @Nullable
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp} or {@code null} if the {@code Span} is still active.
+   *
+   * @return the end {@code Timestamp} or {@code null} if the {@code Span} is still active.
+   */
+  @Nullable
+  public abstract Timestamp getEndTimestamp();
+
+  SpanData() {}
+
+  /** A timed event representation. It can be a timed {@link Annotation} or {@link NetworkEvent}. */
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvent<T> {
+    /**
+     * Returns a new immutable {@code TimedEvent<T>}.
+     *
+     * @param timestamp the {@code Timestamp} of this event.
+     * @param event the event.
+     * @param <T> can be one of {@code Annotation} or {@code NetworkEvent}.
+     * @return a new immutable {@code TimedEvent<T>}
+     */
+    public static <T> TimedEvent<T> create(Timestamp timestamp, T event) {
+      return new AutoValue_SpanData_TimedEvent<T>(
+          checkNotNull(timestamp, "timestamp"), checkNotNull(event, "event"));
+    }
+
+    /**
+     * Returns the {@code Timestamp} of this event.
+     *
+     * @return the {@code Timestamp} of this event.
+     */
+    public abstract Timestamp getTimestamp();
+
+    /**
+     * Returns the event.
+     *
+     * @return the event.
+     */
+    public abstract T getEvent();
+  }
+
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvents<T> {
```

Comment:
in fact add package private constructor on all autovalue types until such time they are supposed to be extended